### PR TITLE
fix(cowSwapper): don't convert orderDigest to Uint8Array / hex

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Web counterpart of https://github.com/shapeshift/hdwallet/pull/635 - open me after it is published

tl;dr is we're currently doing a triple conversion of the `orderDigest` which is useless, as it comes as an hex string already.
While this is theoretically *only* useless back and forth conversions which wouldn't break the flow, unfortunately, we're doing one too much here, meaning this isn't only some benign perf. bottleneck, but an actual product bug.

The conversion is made:

- From an hex string to an Uint8Array in `executeEvmMessage` (see this PR's diff)
- Then in hdwallet's `ethSignMessage`. e.g [WCV2 implementation](https://github.com/shapeshift/hdwallet/blob/9503eeb45085c50324fd93f63a832fe10727ff4f/packages/hdwallet-walletconnectV2/src/ethereum.ts#L86) from a Uint8Array to a Uint8Array (?), then an hex string out of that

https://github.com/shapeshift/hdwallet/pull/635 lays the groundwork for not converting things, but web is the missing bit. This PR fixes things here, by making sure we pass an hexlified string all around.

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes N/A

## Risk

Risk of CoW swaps / ETH messages signing all around. In effect, this means Wherever may be broken across all wallets, and CoW swaps may be broken in other wallets - test accordingly

## Testing

- CoW swaps are working for all wallets
- Wherever signing is working for all wallets

### Engineering

- ☝🏽 

### Operations

- ☝🏽 

## Screenshots (if applicable)

image

https://explorer.cow.fi/gc/orders/0xa476f5b1e4508ad48ad988628a1a68243c08d4a0b9f8a5fe24e68157f86ad1855daf465a9ccf64deb146eeae9e7bd40d6761c986651a9520?tab=overview

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -102,7 +102,7 @@
     "@shapeshiftoss/hdwallet-native": "^1.50.4",
     "@shapeshiftoss/hdwallet-native-vault": "^1.50.4",
     "@shapeshiftoss/hdwallet-shapeshift-multichain": "^1.50.5-alpha.70",
-    "@shapeshiftoss/hdwallet-walletconnectv2": "1.50.8",
+    "@shapeshiftoss/hdwallet-walletconnectv2": "../shapeshiftHdWallet/packages/hdwallet-walletconnectV2",
```
Comment (gomesalexandre) [inline package.json:105]: Install actual package when https://github.com/shapeshift/hdwallet/pull/635 is published

File: src/lib/swapper/swappers/CowSwapper/CowSwapper.ts [hunk 2]
```diff
@@ -27,7 +27,10 @@ export const cowSwapper: Swapper = {
     // Some context about this : https://docs.cow.fi/tutorials/how-to-submit-orders-via-the-api/4.-signing-the-order
     // For more info, check hashOrder method implementation
     const orderDigest = hashOrder(domain(signingDomain, COW_SWAP_SETTLEMENT_ADDRESS), orderToSign)
-    const messageToSign = ethers.utils.arrayify(orderDigest)
+    // orderDigest should be an hex string here. All we need to do is pass it to signMessage/wallet.ethSignMessage and sign it
+    const messageToSign = orderDigest
+    // TODO: signMessage here, as well as ethSignMessage in hdwallet should all expect an hex string as `message` and guard at types/runtime level against
```
Comment (gomesalexandre) [inline src/lib/swapper/swappers/CowSwapper/CowSwapper.ts:32]: Remove TODO when types plumbing is done across web and https://github.com/shapeshift/hdwallet/pull/635

File: src/lib/swapper/swappers/CowSwapper/CowSwapper.ts [hunk 3]
```diff
@@ -27,7 +27,10 @@ export const cowSwapper: Swapper = {
     // Some context about this : https://docs.cow.fi/tutorials/how-to-submit-orders-via-the-api/4.-signing-the-order
     // For more info, check hashOrder method implementation
     const orderDigest = hashOrder(domain(signingDomain, COW_SWAP_SETTLEMENT_ADDRESS), orderToSign)
-    const messageToSign = ethers.utils.arrayify(orderDigest)
+    // orderDigest should be an hex string here. All we need to do is pass it to signMessage/wallet.ethSignMessage and sign it
```
Comment (gomesalexandre) [inline src/lib/swapper/swappers/CowSwapper/CowSwapper.ts:30]: An hex string is, as in hex as a string, not a string as hex - see https://docs.cow.fi/smart-contracts/settlement-contract/signature-schemes
This is effectively an hashed EIP-712 struct

Comment (gomesalexandre) [reply src/lib/swapper/swappers/CowSwapper/CowSwapper.ts:30]: Perhaps https://docs.ethers.org/v5/api/utils/bytes/#utils-isHexString to be safe here

**Comment by 0xApotheosis** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #5382** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5382" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/5382?utm_source=stack-comment).

File: package.json (hunk 1 above)
Comment (0xApotheosis) [reply package.json]: gm https://github.com/shapeshift/web/pull/5382/commits/379dd99b22e62946926978060fa1240891d798d1

**Review by 0xApotheosis — APPROVED** [review]
